src/Data/StringVariants/Util.hs

```diff
@@ -1,19 +1,10 @@
-module Data.StringVariants.Util (natOfLength, useNat, textIsTooLong, textIsWhitespace, textHasNoMeaningfulContent) where
+module Data.StringVariants.Util (natOfLength, useNat, textIsWhitespace, textHasNoMeaningfulContent) where
 import GHC.TypeLits (KnownNat, Nat, SomeNat (..), someNatVal)
 import Prelude
 import Data.Text (Text)
 import qualified Data.Text as T
 import Data.Char (isSpace, isControl)
 
-textIsTooLong :: Text -> Int -> Bool
--- why take? because of stream fusion,
```

Comment:
is the invariant specified in this comment maintained by using compareLength? I *think* it is given that:

> O(n) Compare the count of characters in a Text to a number. Subject to fusion.
>
> This function gives the same answer as comparing against the result of length, but can short circuit if the count of characters is greater than the number, and hence be more efficient.

Reply:
Probably when chessai added this function, compareLength did not exist yet.

Reply:
The function `compareLength` was added in 2010. It's hard to discover, though. The type isn't easy to search on hoogle.